packages/react-components/src/components/chart/contextMenu/ChartContextMenu.tsx

```diff
@@ -1,55 +1,81 @@
-import React from 'react';
-import { Menu, MenuOption } from '../../menu';
+import React, { useEffect, useState } from 'react';
+import { Menu, MenuOptionProps } from '../../menu';
 
 import { CONTEXT_MENU_Z_INDEX } from '../eChartsConstants';
-import { MAX_TREND_CURSORS } from '../trendCursor/constants';
-import { InternalGraphicComponentGroupOption } from '../trendCursor/types';
 
-export type Action = 'add' | 'delete' | 'copy';
+type ContextMenuAction = (offsetX: number) => void;
+type ContextMenuOption = Omit<
```

Comment:
i would suggest to create new types, IMO this makes it super hard to read the code